src/compiler/emitter.ts

```diff
@@ -1179,4394 +1178,5531 @@ namespace ts {
+                    return emitUnparsedSourceOrPrepend(<UnparsedSource>node);
+                case SyntaxKind.UnparsedText:
+                case SyntaxKind.UnparsedInternalText:
+                    return emitUnparsedTextLike(<UnparsedTextLike>node);
+                case SyntaxKind.UnparsedSyntheticReference:
+                    return emitUnparsedSyntheticReference(<UnparsedSyntheticReference>node);
+
+                // Top-level nodes
+                case SyntaxKind.SourceFile:
+                    return emitSourceFile(<SourceFile>node);
+                case SyntaxKind.Bundle:
+                    Debug.fail("Bundles should be printed using printBundle");
+                    break;
+                // SyntaxKind.UnparsedSource (handled above)
+                case SyntaxKind.InputFiles:
+                    Debug.fail("InputFiles should not be printed");
+                    break;
+
+                // JSDoc nodes (only used in codefixes currently)
+                case SyntaxKind.JSDocTypeExpression:
+                    return emitJSDocTypeExpression(node as JSDocTypeExpression);
+                case SyntaxKind.JSDocNameReference:
+                    return emitJSDocNameReference(node as JSDocNameReference);
+                case SyntaxKind.JSDocAllType:
+                    return writePunctuation("*");
+                case SyntaxKind.JSDocUnknownType:
+                    return writePunctuation("?");
+                case SyntaxKind.JSDocNullableType:
+                    return emitJSDocNullableType(node as JSDocNullableType);
+                case SyntaxKind.JSDocNonNullableType:
+                    return emitJSDocNonNullableType(node as JSDocNonNullableType);
+                case SyntaxKind.JSDocOptionalType:
+                    return emitJSDocOptionalType(node as JSDocOptionalType);
+                case SyntaxKind.JSDocFunctionType:
+                    return emitJSDocFunctionType(node as JSDocFunctionType);
+                case SyntaxKind.RestType:
+                case SyntaxKind.JSDocVariadicType:
+                    return emitRestOrJSDocVariadicType(node as RestTypeNode | JSDocVariadicType);
+                // SyntaxKind.JSDocNamepathType (missing)
+                case SyntaxKind.JSDocComment:
+                    return emitJSDoc(node as JSDoc);
+                case SyntaxKind.JSDocTypeLiteral:
+                    return emitJSDocTypeLiteral(node as JSDocTypeLiteral);
+                case SyntaxKind.JSDocSignature:
+                    return emitJSDocSignature(node as JSDocSignature);
+                case SyntaxKind.JSDocTag:
+                case SyntaxKind.JSDocClassTag:
+                    return emitJSDocSimpleTag(node as JSDocTag);
+                case SyntaxKind.JSDocAugmentsTag:
+                case SyntaxKind.JSDocImplementsTag:
+                    return emitJSDocHeritageTag(node as JSDocImplementsTag | JSDocAugmentsTag);
+                // SyntaxKind.JSDocAuthorTag (missing)

```

Comment:
Should we open issues for the missing emit node kinds? I'm not sure under what circumstances we emit jsdoc using the emitter, but when we do, I assume these node kinds being missing maybe has an effect?

Reply:
We only emit JSDoc tags using the emitter as part of refactors. I assume we've only been adding them on an as-need basis.